File: app/src/main/java/com/boost/espressotest/data/di/StorageModule.java
package com.boost.espressotest.data.di;

import android.arch.persistence.room.Room;

import com.boost.espressotest.app.EspressoTestApp;
import com.boost.espressotest.data.local.AppDatabase;
import com.boost.espressotest.data.local.dao.ProductDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author PerSpiKyliaTor on 22.01.18.
 */

@Module
public class StorageModule {

    @Provides
    @Singleton
    AppDatabase provideAppDatabase(EspressoTestApp espressoTestApp) {
        return Room.databaseBuilder(espressoTestApp.getApplicationContext().getApplicationContext(),
                AppDatabase.class, AppDatabase.APP_DATABASE_NAME).build();
    }

    @Provides
    @Singleton
    ProductDao provideProductDao(AppDatabase appDatabase) {
        return appDatabase.productDao();
    }
}

File: app/src/main/java/com/boost/espressotest/data/di/StorageModule.kt
package com.boost.espressotest.data.di

import android.content.Context
import androidx.room.Room
import com.boost.espressotest.data.local.AppDatabase
import com.boost.espressotest.data.local.dao.ProductDao
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class StorageModule {

    companion object {
        const val APP_DATABASE_NAME = "app-database"
    }

    @Provides
    @Singleton
    internal fun provideAppDatabase(context: Context): AppDatabase {
        return Room.databaseBuilder(context, AppDatabase::class.java, APP_DATABASE_NAME).build()
    }

    @Provides
    @Singleton
    internal fun provideProductDao(appDatabase: AppDatabase): ProductDao {
        return appDatabase.productDao()
    }
}
